Cargo.toml

```diff
@@ -24,15 +24,16 @@ rnix = "0.10.2"
 rowan = "0.12.6"
 serde = "1.0.104"
 serde_json = "1.0.44"
-nixpkgs-fmt-rnix = "1.2.0"
+nixpkgs-fmt-rnix = { version = "1.2.0", optional = true }
+alejandra = { version = "1.5.0", optional = true }
```

Comment:
I'm not familiar with the LSP, but I can throw here that there is a third approach: using an environment variable pointing to the formatter binary, and just feeding the code via stdin and getting the output via stdout. That way the closure would decrease substantially as we would be reusing the binaries already present in the user system (instead of compiling and linking them here), we would not need to implement a configuration handler for this LSP, and we would unify the interface and allow for more formatters to be chosen in the future (including `nix fmt`)

Just an idea, you both are more familiar with this project than me

Reply:
> Of course, if more formatters emerge (https://xkcd.com/927/) that could become an issue

Well, the primary reason why I even consider this is because Alejandra seems to be used by quite a bunch of people and is IIRC even a candidate for the nixpkgs formatting RFC, so integrating it is justifiable IMHO.

> That way the closure would decrease substantially as we would be reusing the binaries already present in the user system

Admittedly I haven't checked the actual impact on closure size of both alejandra and nixpkgs-fmt, but an LSP server is something that's only running on developer machines anyways, so I wouldn't consider a slight (i.e. non-optimal, but not "bad") increase of closure-size much of an issue.

> I'm not familiar with the LSP, but I can throw here that there is a third approach: using an environment variable pointing to the formatter binary, and just feeding the code via stdin and getting the output via stdout.

As described in #87 partial reformats are desirable and something I'd like to see here as well. If we get somewhat reasonable APIs from formatters for that, I'd prefer that solution over having to parse stdout from a formatter. I could imagine that we'd also run into funny quoting or encoding issues when reformatting files with weird stuff in it and parse the (structured) output of the (incremental) edits from stdout.